core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataEditorPanelService.js

```diff
@@ -137,23 +145,19 @@ oppia.factory('TrainingDataEditorPanelService', [
                 // classification results to the creator.
                 if (classificationType === EXPLICIT_CLASSIFICATION ||
                     classificationType === TRAINING_DATA_CLASSIFICATION) {
-                  $scope.newAnswerIsResolved = true;
+                  $scope.newAnswerIsAlreadyResolved = true;
                 } else {
-                  TrainingDataService.trainAnswerGroup(
+                  TrainingDataService.associateWithAnswerGroup(
                     answerGroupIndex, newAnswer);
                   _rebuildTrainingData();
                   $scope.answerSuccessfullyAdded = true;
                 }
               };
 
               $scope.openTrainUnresolvedAnswerModal = function(answerIndex) {
-                answer = $scope.trainingData[answerIndex].answer;
+                var answer = $scope.trainingData[answerIndex].answer;
                 return TrainingModalService.openTrainUnresolvedAnswerModal(
-                  answer, true, $scope.finishTrainingAnswer);
-              };
-
-              $scope.finishTrainingAnswer = function() {
-                _rebuildTrainingData();
+                  answer, true, _rebuildTrainingData());
```

Comment:
Did you test this (at least manually)?

Reply:
Ah, yes. I did but I missed this scenario. Thanks for notifying.